File: packages/shodoukan/src/shodoukan/models/kanji.py
from pydantic import BaseModel


class KanjiMeaning(BaseModel):
    text: str
    lang: str


class Kanji(BaseModel):
    literal: str
    grade: int | None
    stroke_count: int
    freq: int | None
    jlpt: int | None
    on_readings: list[str]
    kun_readings: list[str]
    nanori: list[str]
    meanings: list[KanjiMeaning]

File: packages/shodoukan-web/models/kanji.ts
export interface KanjiMeaning {
  text: string
  lang: string
}

export interface Kanji {
  literal: string
  grade: number | null
  stroke_count: number
  freq: number | null
  jlpt: number | null
  on_readings: string[]
  kun_readings: string[]
  nanori: string[]
  meanings: KanjiMeaning[]
}

export interface Language {
  code: string
  label: string
}

export const SUPPORTED_LANGUAGES: Language[] = [
  { code: 'en', label: 'English' },
  { code: 'es', label: 'Español' },
  { code: 'fr', label: 'Français' },
  { code: 'de', label: 'Deutsch' },
]
